### PR TITLE
Checkpoint-03-Habtamu Abere

### PR DESCRIPTION
**Pull request by HabRonan**
comfort-4.5
completeness-4.8

### CODE REVIEW

File: oojs.js [hunk 1]
```diff
@@ -10,3 +10,45 @@
   // Have an artist property that is determined by some input.
 
 // Create an instance of the Album class.
+
+//defining a playlist class
+ class playlist{
```
Comment (amaseda) [inline oojs.js:15]: :+1: This works but you should always capitalize the first letter of a class name.

File: oojs.js [hunk 2]
```diff
@@ -10,3 +10,45 @@
...
+   constructor(title){
+     this.title = title;
+     this.songs = [];
```
Comment (amaseda) [inline oojs.js:18]: The instructions say `songs` is supposed to be determined by a user input

File: oojs.js [hunk 3]
```diff
@@ -10,3 +10,45 @@
...
+     this.favorites = 0;
```
Comment (amaseda) [inline oojs.js:19]: :+1:

File: oojs.js [hunk 4]
```diff
@@ -10,3 +10,45 @@
...
+
+   }
+// adding a method that adds a song to the playlist
+   addSong(songtitle){
+      this.songtitle = songtitle;
```
Comment (amaseda) [inline oojs.js:24]: You don't need to attach the song title to the playlist using `this.songTitle`. The important thing is to just push a song into `this.songs`

File: oojs.js [hunk 5]
```diff
@@ -10,3 +10,45 @@
...
+//defining a playlist class
+ class playlist{
+   constructor(title){
+     this.title = title;
+     this.songs = [];
+     this.favorites = 0;
+
+   }
+// adding a method that adds a song to the playlist
+   addSong(songtitle){
```
Comment (amaseda) [inline oojs.js:23]: :+1:

File: oojs.js [hunk 6]
```diff
@@ -10,3 +10,45 @@
...
+      this.songtitle = songtitle;
+      this.songs.push(`${this.songtitle}`);
+     }
+
+ }
+//creat an instance of the playlist class
+  let  playlist1  = new playlist("rock");
```
Comment (amaseda) [inline oojs.js:30]: :+1: Correct but `songs` is supposed to be a user input, so would need to include that here as well.

File: oojs.js [hunk 7]
```diff
@@ -10,3 +10,45 @@
...
+  let playlist2 = new playlist("pop");
+
+  console.log(playlist1);
+  console.log(playlist2);
+  console.log(this.songs);
```
Comment (amaseda) [inline oojs.js:35]: Calling `this` in the global scope will equal the browser `Window` object -- not the class instance.

File: oojs.js [hunk 8]
```diff
@@ -10,3 +10,45 @@
...
+   playlist1.addSong("burning love");
+
+    playlist2.addSong("dangerous");
+    playlist2.addSong("beatit");
+    playlist2.addSong("Thriller");
+
+    class album extends playlist{
```
Comment (amaseda) [inline oojs.js:42]: :+1: but again, remember capitalization.

File: oojs.js [hunk 9]
```diff
@@ -10,3 +10,45 @@
...
+
+           constructor(title,artist){
+             super(title);
```
Comment (amaseda) [inline oojs.js:45]: :+1: but would need to account for `songs` here

File: fp.js [hunk 10]
```diff
@@ -6,10 +6,16 @@ var people = [
   {name: "Jessica", age: 34},
   {name: "Jasmine", age: 39}
 ]
+ /*
 people.forEach(function(person){
   peoplesAges.push(person.age)
 });
-
+*/
+  people.map(people =>{
```
Comment (amaseda) [inline fp.js:14]: `.map` returns a new array containing a collection of things that match its `return` statement. In this case, you want to use map to generate an array containing only names, so the solution looks something like this...

```js
var people = {
  // data goes here
}

var peopleAges = people.map(person => {
  return person.age
})
```

This avoids the need to initialize an empty array and have to push items into it.

File: fp.js [hunk 11]
```diff
@@ -18,17 +24,36 @@ var people = [
...
   {name: "Jasmine", age: 39}
 ]
+/*
 people.forEach(function(person) {
   if(person.age < 35){
     peopleYoungerThan35.push(person)
   }
 });
+*/
+ people.filter(people=>{
```
Comment (amaseda) [inline fp.js:34]: Like `.map`, with `.filter` you do not need to push into an empty array. Instead, you can save the result of filter directly in a variable. Filter's return statement should be whatever condition you want all the items in the resulting array to match. So the solution would look something like this...

```js
var peopleYoungerThan35 = people.filter(person => {
  return person.age < 35  // this means we want to only return items in `people` that have an age under 35
})
```

File: fp.js [hunk 12]
```diff
@@ -18,17 +24,36 @@ var people = [
...
   {name: "Jane", age: 32},
   {name: "Jessica", age: 34},
   {name: "Jasmine", age: 39}
 ]
-var peopleWhoseNamesHaveTheLetterS = people.filter(function(person){return person.name.match("s")});
+  //var peopleWhoseNamesHaveTheLetterS = people.filter(function(person){return person.name.match("s")});
+
+    people.forEach(people=>{
+    if(people.name.match("s")){
+      return peopleWhoseNamesHaveTheLetterS.push(people);
```
Comment (amaseda) [inline fp.js:54]: :+1: Although you don't need to use the `return` keyword here

Also, you should not use `people` as the argument for your callback. Instead, use something that represents each item in the `people` array -- like `person`.

File: oojs.js (hunk 1 above)
Comment (HabRonan) [reply oojs.js:15]: Thanks

File: oojs.js (hunk 2 above)
Comment (HabRonan) [reply oojs.js:18]: right. Thanks

File: oojs.js (hunk 7 above)
Comment (HabRonan) [reply oojs.js:35]: Oh ya, my bad.

**Comment by HabRonan** [comment]
Thanks for the comments. I will check everything again.

**Pull request closed without merging**